README


This archive contains the distribution KiokuDB-Backend-DBI,
version 1.22:

  L<DBI> backend for L<KiokuDB>

This software is copyright (c) 2012 by Yuval Kogman, Infinity Interactive.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
1.22
    - Fix lookup of invalid entries that differ from a valid entry only in case

1.21
    - Fix undocumented gin_index option to define_kiokudb_schema (frew)

1.20
    - Fix some stuff with recent DBIx::Class (arcanez)

1.19
	- Merge releases

1.16 (NUFFIN)
    - Updated to support change in DBIC object hierarchy

1.18
    - Fix set subquery building. Bad use of parenthesis.
    - Includes an actual test for the set query.

1.17
    - Small fix in release process

1.16 (DRUOSO)
    - Adds support for Set GIN queries

1.15
    - Constructing a KiokuDB directory handle is no longer mandatory, you can
      get it from the DBIC schema (domm, nothingmuch)

1.14
    - documentation fixes

1.13
    - fix ordered loading of DBIC Core component in example code
    - FOR UPDATE added to SELECT queries on MySQL, PG and Oracle
    - compound operations now create a txn guard by default

1.12
    - Fix FLORA's botched upload ;-) (typo in dist.ini)

1.11
    - Support for DBIx::Class integration (KiokuDB objects and DBIC rows can
      point to each other, DBIC resultsets can be serialized in KiokuDB)
    - All SQL operations that use placeholders now support limiting of the
      number of placeholders, splitting the operation into batches (this is
      enabled by default for SQLite which limits SQL statements to 999
      placeholders) (Jason May)
    - DBIx::Class::Optional::Dependencies now used in Makefile.PL to avoid
      problems with 'create => 1'

1.10
    - NOTE: this was supposed to be 0.10... oops
    - Added mysql_strict attribute, true by default, causes SQL's strict mode
      to be set
    - Now specifies 'longblob' instead of 'blob' as the type for the 'data'
      column when deploying to MySQL because the "large" in blob apparently
      means 64k and MySQL was truncating it by default
    - Added more bitter comments about MySQL to the documentation
    - Make the table_info fix from 0.09 only apply to SQLite

0.09
    - Fix double deployment with recent versions of DBD::SQLite
    - Pod fixes

0.08
    - Fix GIN key update (FIXME no coverage in KiokuDB test suite yet)
    - uses done_testing in test suite

0.07
    - Various documentation fixes
    - ID stream optimization
    - use the new sqlt_deploy_callback
    - Don't use execute_array for DELETE
    - Explicitly call $schema->disconnect on destruction
    - Allow coderef in DBI DSN
    - Don't reuse DBIC's txn_do

0.06
    - Skip tests if SQL::Translator is missing

0.05
    - Use an update statement when an entry the 'prev' attribute, and insert
      otherwise.
    - Add a 'schema_hook' attribute, to allow modification of the cloned schema
      before connecting

0.04
    - Remove DBD::Pg usage (DBIC handles those parts now)

0.03
    - Switch to Serialize::Delegate
    - add 'create' attribute

0.02
    - Skip tests if DBD::SQLite is not available

0.01
    - Initial release


Makefile.PL
# This Makefile.PL for  was generated by Dist::Zilla.
# Don't edit it but the dist.ini used to construct it.

use strict;
use warnings;
use ExtUtils::MakeMaker 6.30;

my %WriteMakefileArgs = (
  'ABSTRACT' => 'L<DBI> backend for L<KiokuDB>',
  'AUTHOR' => 'Yuval Kogman',
  'BUILD_REQUIRES' => {
    'DBD::SQLite' => '0',
    'DBI' => '0',
    'KiokuDB::Test' => '0',
    'Search::GIN::Extract::Callback' => '0',
    'Search::GIN::Extract::Class' => '0',
    'Search::GIN::Query::Manual' => '0',
    'Search::GIN::Query::Set' => '0',
    'Test::Exception' => '0',
    'Test::More' => '0',
    'Test::TempDir' => '0',
    'YAML::XS' => '0',
    'constant' => '0',
    'ok' => '0'
  },
  'CONFIGURE_REQUIRES' => {
    'DBIx::Class::Optional::Dependencies' => '0',
    'ExtUtils::MakeMaker' => '6.30'
  },
  'DISTNAME' => 'KiokuDB-Backend-DBI',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'KiokuDB::Backend::DBI',
  'PREREQ_PM' => {
    'Carp' => '0',
    'Class::Accessor::Grouped' => '0',
    'DBI' => '1.607',
    'DBIx::Class' => '0.08127',
    'DBIx::Class::Core' => '0',
    'DBIx::Class::ResultSource::Table' => '0',
    'DBIx::Class::Schema' => '0',
    'Data::Stream::Bulk' => '0.07',
    'Data::Stream::Bulk::DBI' => '0',
    'JSON' => '0',
    'KiokuDB' => '0.46',
    'KiokuDB::Backend' => '0',
    'KiokuDB::Backend::Role::Clear' => '0',
    'KiokuDB::Backend::Role::Concurrency::POSIX' => '0',
    'KiokuDB::Backend::Role::GC' => '0',
    'KiokuDB::Backend::Role::Query::GIN' => '0',
    'KiokuDB::Backend::Role::Query::Simple' => '0',
    'KiokuDB::Backend::Role::Scan' => '0',
    'KiokuDB::Backend::Role::TXN' => '0',
    'KiokuDB::Backend::Serialize::Delegate' => '0',
    'KiokuDB::TypeMap' => '0',
    'KiokuDB::TypeMap::Entry' => '0',
    'KiokuDB::TypeMap::Entry::Naive' => '0',
    'List::MoreUtils' => '0',
    'Moose' => '0',
    'Moose::Util::TypeConstraints' => '0',
    'MooseX::Types' => '0.08',
    'MooseX::Types::Moose' => '0',
    'SQL::Abstract' => '0',
    'Scalar::Util' => '0',
    'Search::GIN' => '0.07',
    'Search::GIN::Extract::Delegate' => '0',
    'Test::More' => '0.88',
    'Test::TempDir' => '0',
    'Test::use::ok' => '0',
    'Try::Tiny' => '0',
    'base' => '0',
    'namespace::autoclean' => '0',
    'namespace::clean' => '0',
    'strict' => '0',
    'warnings' => '0'
  },
  'VERSION' => '1.22',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

require DBIx::Class::Optional::Dependencies;

$WriteMakefileArgs{PREREQ_PM} = {
    %{ $WriteMakefileArgs{PREREQ_PM} || {} },
    %{ DBIx::Class::Optional::Dependencies->req_list_for ('deploy') },
};

WriteMakefile(%WriteMakefileArgs);

